lib/json_schemer/schema/base.rb

```diff
@@ -298,6 +298,17 @@ def validate_ref(instance, ref, &block)
             parent_uri: ref_uri
           )
           validate_instance(subinstance, &block)
+        elsif ref_uri.to_s.include?("##{ref_uri.fragment}")
```

Comment:
Is this testing if there is a fragment?  If so, the condition could be `ref_uri.fragment`.

Reply:
Yes. Hana above does not think this a valid fragment (because it does not start with `/`) so we need a new conditional to capture this. Fragment returns "bar" but what we really want to make sure if "bar" is a pointer so ensuring that that the ref_uri includes a valid id pointer("#bar")